src/pudl/extract/xbrl.py

```diff
@@ -0,0 +1,166 @@
+"""Generic extractor for all FERC XBRL data."""
+import io
+import json
+import time
+from datetime import datetime
+
+import sqlalchemy as sa
+from ferc_xbrl_extractor import xbrl
+from ferc_xbrl_extractor.instance import InstanceBuilder
+
+import pudl
+from pudl.helpers import get_logger
+from pudl.settings import FercGenericXbrlToSqliteSettings, FercToSqliteSettings
+from pudl.workspace.datastore import Datastore
+
+logger = get_logger(__name__)
+
+XBRL_FORM_NUMBERS = [1, 2, 6, 60, 714]
+
+
+class FercXbrlDatastore:
+    """Simple datastore wrapper for accessing ferc1 xbrl resources."""
+
+    def __init__(self, datastore: Datastore):
+        """Instantiate datastore wrapper for ferc1 resources."""
+        self.datastore = datastore
+
+    def get_filings(self, year: int, form_number: int):
+        """Return list of filings from archive."""
+        archive = self.datastore.get_zipfile_resource(f"ferc{form_number}", year=year)
+
+        # Load RSS feed metadata
+        filings = []
+        with archive.open("rssfeed") as f:
+            metadata = json.load(f)
+
+            # Loop through all filings by a given filer in a given quarter
+            # And take the most recent one
+            for key, filing_info in metadata.items():
+                latest = datetime.min
+                for filing_id, info in filing_info.items():
+                    # Parse date from 9-tuple
+                    published = datetime.fromtimestamp(
+                        time.mktime(tuple(info["published_parsed"]))
+                    )
+
+                    if published > latest:
+                        latest_filing = f"{filing_id}.xbrl"
+
+                # Create in memory buffers with file data to be used in conversion
+                filings.append(
+                    InstanceBuilder(
+                        io.BytesIO(archive.open(latest_filing).read()), filing_id
+                    )
+                )
+
+        return filings
+
+
+def _get_sqlite_engine(
+    form_number: int, pudl_settings: dict, clobber: bool
+) -> sa.engine.Engine:
+    """Create SQLite engine for specified form and drop tables.
+
+    Args:
+        form_number: FERC form number.
+        pudl_settings: Dictionary of settings.
+        clobber: Flag indicating whether or not to drop tables.
+    """
+    # Read in the structure of the DB, if it exists
+    logger.info(f"Dropping the old FERC Form {form_number} SQLite DB if it exists.")
```

Comment:
Could you also mention that this only pertains to the XBRL database in this message? Otherwise it's scary and I think my big FERC Form 1 DBF derived SQLite DB is getting wiped.

Reply:
Fixed